test/gurka/test_search_filter.cc

```diff
@@ -61,6 +61,23 @@ TEST_F(SearchFilter, Unfiltered) {
   gurka::assert::osrm::expect_steps(result, {"AB", "BC"});
   gurka::assert::raw::expect_path(result, {"AB", "BC"});
 }
+TEST_F(SearchFilter, NodeSnapped) {
+  auto from = "B";
+  auto to = "C";
```

Comment:
b to c is a tunnel way, so if we node snap to b, the edge bc should not be allowed as a candidate. before the fix it was and this test failed. with the update the test passes and the route goes around the tunnel.